v-next/hardhat/src/internal/builtin-plugins/network-manager/config-resolution.ts

```diff
@@ -291,3 +301,15 @@ export function resolveInitialBaseFeePerGas(
     ? BigInt(initialBaseFeePerGas)
     : undefined;
 }
+
+function resolveDefaultBlockGasLimit(
+  hardfork: string,
+  chainType: ChainType | undefined = GENERIC_CHAIN_TYPE,
+): bigint {
+  const isPostOsaka =
+    chainType === OPTIMISM_CHAIN_TYPE
+      ? false // TODO: implement OP hardfork comparison once we have more OP hardforks
+      : hardforkGte(hardfork, "osaka", chainType);
+
+  return isPostOsaka ? 16_777_216n : 30_000_000n;
```

Comment:
The Osaka hardfork introduced a transaction gas cap. This makes the default block gas limit dependant on the hardfork. When the hardfork is post-Osaka, we use the transaction gas cap.